datacube/api/core.py

```diff
@@ -532,6 +534,21 @@ def __exit__(self, type_, value, traceback):
         self.close()
 
 
+def _validate_geobox(geobox):
```

Comment:
This makes a few assumptions that might bite someone down the line, if there are CRSes using strange units, or we want to do super high resolution stuff. Might be good to add a docstring stating these assumptions. eg.
> Ensure that a geobox has a reasonable size and resolution
> 
> Where 'reasonable' means that each pixel is larger than about 10cm^2 and smaller than about 1000km^2. This may  fail if a CRS uses units other than metres.